cmd/osbuild-jobsite-manager/main.go

```diff
@@ -264,13 +268,68 @@ func StepProvision(manifest []byte) error {
 
 func StepPopulate() error {
 	return Wait(argTimeoutPopulate, func(done chan<- struct{}, errs chan<- error) {
-		res, err := Request("POST", "populate", []byte(""))
+		file, err := os.CreateTemp(filepath.Dir(argStore), "store.*.tar")
+		if err != nil {
+			errs <- err
+			return
+		}
+		tw := tar.NewWriter(file)
+		defer tw.Close()
+		err = filepath.Walk(argStore, func(filePath string, fi os.FileInfo, err error) error {
+			if err != nil {
+				return err
+			}
+
+			// skip the store direction itself, just package the contents
+			if filePath == argStore {
+				return nil
+			}
+
+			if !fi.Mode().IsRegular() && !fi.Mode().IsDir() {
+				return nil
+			}
+			header, err := tar.FileInfoHeader(fi, fi.Name())
+			if err != nil {
+				return err
+			}
+
+			// FileInfo only contains the basename, see https://pkg.go.dev/archive/tar#FileInfoHeader.
+			header.Name = strings.TrimPrefix(strings.Replace(filePath, argStore, "", -1), string(filepath.Separator))
+			if err := tw.WriteHeader(header); err != nil {
+				return err
+			}
 
+			if fi.Mode().IsRegular() {
```

Comment:
Error if it isn't?

Reply:
directories just need the header :) so this is fine I think.